core/state_processor.go

```diff
@@ -81,6 +81,9 @@ func (p *StateProcessor) Process(block *types.Block, statedb *state.StateDB, cfg
 		statedb.IntermediateRoot(true)
 	}
 
+	parentHeader := p.bc.GetHeaderByNumber(block.NumberU64() - 1)
+	parentVmRunner := p.bc.NewEVMRunner(parentHeader, statedb)
```

Comment:
I'm not 100% sure of this and would want to test it, but I don't think this is correct without a `statedb.Copy()` or performing all of the contract calls when creating the `SysContractCallCtx` (actually also my preference). The statedb is a shared object and is being updated during block execution.

Reply:
> or performing all of the contract calls when creating the SysContractCallCtx (actually also my preference).

Why do you prefer it?

Reply:
Under the assumption that there will be consistent usage of the different fields, it avoids a conditional check later on. Only when you don't use all of the data does it make sense to delay computation. Under our current load, it's probably fine either way (excepting the statedb is mutable state issue)

Reply:
Sounds good. I'll keep current implementation for the reason that we might add more fields to `SysContractCallCtx` in the future. So I fixed the use of `stateDB`, by using `stateDB.Copy()`, both in `stateProcessor.Process()` and `BlockState`.